orchagent/portsorch.cpp

```diff
@@ -2524,6 +2554,18 @@ bool PortsOrch::setGearboxPortAttr(const Port &port, dest_port_type_t port_type,
                     m_gearboxTable->hset(key, speed_attr, to_string(speed));
                     SWSS_LOG_NOTICE("BOX: Updated APPL_DB key:%s %s %d", key.c_str(), speed_attr.c_str(), speed);
                 }
+                else if (id == SAI_PORT_ATTR_FEC_MODE && fec_override_sup)
```

Comment:
could you please elaborate on why we need this?

Reply:
According to the existing behavior, if the user FEC is configured explicitly, it should override even when autoneg is turned on. Without SAI library supporting override attribute, this assumption was implicit. When SAI library supports override attribute, the override needs to be explicitly programmed.